app/models/concerns/message_search_data.rb

```diff
@@ -0,0 +1,103 @@
+module MessageSearchData
```

Comment:
Can you check the Conversations::EventDataPresenter and try implementing a similar pattern (eg: Message::SearchDataPresenter) and add a spec so that it is easier to test it?

Reply:
Ah, I need to learn where things plug in.

I've made this change.